modules/angular-meteor-collection.js

```diff
@@ -184,87 +231,107 @@ angularMeteorCollection.factory('AngularMeteorCollection', [
           }
         }
       });
+  
+      this._hDataAutorun = Tracker.autorun(function() {
+        cursor.fetch();
+        if (serverMode) {
+          unsetServerUpdateMode();
+        }
+      });
     };
 
-    // Function applies multiple operations (savings, deletions etc) once
-    // with the help of $timeout.
-    AngularMeteorCollection._safeApply = function(promise) {
-      var self = this;
+    AngularMeteorCollection.stop = function() {
+      this._stopCursor();
+      this._hNewCurAutorun.stop();
+    };
 
-      if (promise) {
-        $timeout.cancel(promise);
-      }
+    AngularMeteorCollection._stopCursor = function() {
+      this._unsetAutoClientSave();
 
-      // Clearing the watch is needed so no updates are sent to server
-      // while handling updates from the server.
-      if (!this.UPDATING_FROM_SERVER) {
-        this.UPDATING_FROM_SERVER = true;
-        if (!$rootScope.$$phase) $rootScope.$apply();
+      if (this._hObserve) {
+        this._hObserve.stop();
+        this._hDataAutorun.stop();
       }
 
-      return $timeout(function () {
-        // Saves changes happened within the previous update from server.
-        collectionUtils.updateCollection(self, self._serverBackup, self.diffArrayFunc);
-        self.UPDATING_FROM_SERVER = false;
-      }, 50);
+      this.splice(0);
+      this._serverBackup.splice(0);
     };
 
-    AngularMeteorCollection.stop = function () {
-      if (this._unregisterAutoBind) {
-        this._isAutoBind = false;
-        this._unregisterAutoBind();
+    AngularMeteorCollection._unsetAutoClientSave = function(name) {
+      if (this._hRegAutoBind) {
+        this._hRegAutoBind();
+        this._hRegAutoBind = null;
       }
+    };
 
-      if (this.observeHandle)
-        this.observeHandle.stop();
+    AngularMeteorCollection._setAutoClientSave = function() {
+      var self = this;
 
-      this.splice(0);
-      this._serverBackup.splice(0);
+      // Always unsets auto save to keep only one $watch handler. 
```

Comment:
Have you tested against the scenario of which I've added the _isAutobind flag?  this is a complex scenario